pkg/app/accounts/usernameonly.go

```diff
@@ -38,8 +38,12 @@ func NewUsernameOnlyAccountManager() *UsernameOnlyAccountManager {
 	return &UsernameOnlyAccountManager{}
 }
 
-func (m *UsernameOnlyAccountManager) UserFromRequest(r *http.Request) (User, error) {
-	return userFromRequest(r)
+func (m *UsernameOnlyAccountManager) UserFromRequest(w http.ResponseWriter, r *http.Request) (User, error) {
```

Comment:
let's avoid modifying the `Manager` interface in this case.  We can add the redirection logic directly in the handler here: https://github.com/google/cloud-android-orchestration/blob/main/pkg/app/app.go#L517, if `user == nil` return a typed error, catch the error in the handler in redirect from there. here's a good error about errors and errors handling in golang: https://go.dev/blog/error-handling-and-go

Reply:
Thanks for sharing the Golang good practices, super helpful!

I'm wondering if it would be better to delegate the trigger of redirection to `UsernameOnlyAccountManager`, because that only this account manager type needs that. So I would like to return this new error type at `UsernameOnlyAccountManager` instead of when `app.go` receives a `nil` user. What do you think?

Reply:
yes, you got it. the new error should be returned from here in `UsernameOnlyAccountManager.UserFromRequest`.